egs2/TEMPLATE/asr1/asr.sh

```diff
@@ -278,6 +282,21 @@ fi
 
 
 if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
+    if "${speed_perturb}" && [ -n "${perturb_factors}" ]; then
+        # speed-perturbed
+        temp_files=""
+        num_factors=0
+        for factor in ${perturb_factors}; do
+            num_factors=$((num_factors+1))
+            temp_files+=" data/temp${num_factors}"
+            utils/perturb_data_dir_speed.sh ${factor} data/train data/temp${num_factors}
```

Comment:
Thank you for adding speed perturbation, but I need to point small things: 

- Would you create it as new stage? i.e. 2nd stage is speed perturb and 3rd is feature extract. 
   - (I think there are no disadvantages to split stages into small pieces.)
- utils/perturb_data_dir_speed.sh  depends on `wav-to-duration` to create utt2dur or reco2dur, but we are planning to make espnet as kaldi-free.
   - In my understanding, It's not required to generate utt2dur and reco2dur for after stages, so It's enough to create new `utils/perturb_data_dir_speed.sh` without invoking ` utils/data/get_utt2dur.sh`
- The for-block to handle ${perturb_factors} looks a little bit complex, how about like this?
   - I think we don't need to remove intermediate directory,

```bash
if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
    if [ -n "${speed_perturb_factors}" ]; then
       log "stage 2: Speed perturbation: data/${train_set} -> data/${train_set}_sp"
       for factor in ${speed_perturb_factors}; do
           scripts/utils/perturb_data_dir_speed.sh "${speed_perturb_factors}"  "data/${train_set}" "data/${train_set}_sp${factor}"
           _dirs+="data/${train_set}_sp${factor}"
       done
       utils/combine_data.sh "data/${train_set}"  ${_dirs}
    else
       log "Skip stage 2: Speed perturbation"
    fi
fi

if [ -n "${speed_perturb_factors}" ]; then
    train_set="${train_set}_sp"
fi 

if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ]; then
    log "Stage 3: Format wav.scp: data/ -> ${data_feats}/org/"
```

Actually this is my TODO, so you can keep the current shape if you don't have time.
Do you think I should do this or would you do?

Reply:
Thanks! I will add speed perturbation following your suggestions.